### PR TITLE
Add long error code explanation message for E0203

### PR DESCRIPTION
**Pull request by aDotInTheVoid**
Addressed some of #61137

r? @GuillaumeGomez

### CODE REVIEW

File: src/librustc_error_codes/error_codes/E0203.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Having multiple relaxed default bounds is unsuported.
```
Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0203.md]: "unsupported"

File: src/librustc_error_codes/error_codes/E0203.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+Erroneous code example:
+
+```compile_fail,E0203
+
```
Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0203.md]: Please remove this line.

File: src/librustc_error_codes/error_codes/E0203.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+trait Foo {}
+
+struct S5<T>(*const T) where T: ?Foo + ?Sized;
+```
+
+Here the type `T` cannot have a relaxed bound for both `Foo` and `Sized`
```
Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0203.md]: Missing a dot at the end of the sentence.

**Comment by GuillaumeGomez** [comment]
Looks good globally. Just a few little things missing. Can you extend your explanation of the error a bit more and also add a working example (the failing code fixed) please? Once done, I'll review one last time and then I think it'll be good to go!

**Comment by GuillaumeGomez** [comment]
Sorry, closed by mistake...

File: src/librustc_error_codes/error_codes/E0203.md (hunk 1 above)
Comment (aDotInTheVoid) [reply src/librustc_error_codes/error_codes/E0203.md]: Fixed

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**